src/app/inventory/store/stats.ts

```diff
@@ -389,46 +389,102 @@ function enhanceStatsWithPlugs(
   // stats in first.
   const sortedSockets = _.sortBy(createdItem.sockets.allSockets, (s) => s.plugOptions.length);
   for (const socket of sortedSockets) {
-    for (const plug of socket.plugOptions) {
-      if (plug.plugDef.investmentStats?.length) {
-        plug.stats = buildPlugStats(plug, statsByHash, statDisplays);
-      }
-    }
+    buildPlugStats(socket, statsByHash, statDisplays);
   }
 }
 
 /**
- * For each stat this plug modified, calculate how much it modifies that stat.
- *
- * Returns a map from stat hash to stat value.
+ * Attaches a stats attribute to each plug in the socket which is an object of statTypeHash to the value it
+ * will contribute to the items stats when socketed.
  */
 function buildPlugStats(
-  plug: DimPlug,
+  socket: DimSocket,
   statsByHash: { [statHash: number]: DimStat },
   statDisplays: { [statHash: number]: DestinyStatDisplayDefinition }
 ) {
-  const stats: {
-    [statHash: number]: number;
-  } = {};
-
-  for (const perkStat of plug.plugDef.investmentStats) {
-    let value = perkStat.value;
-    const itemStat = statsByHash[perkStat.statTypeHash];
-    const statDisplay = statDisplays[perkStat.statTypeHash];
-    if (itemStat && statDisplay) {
-      // This is a scaled stat, so we need to scale it in context of the original investment stat.
-      // Figure out what the interpolated stat value would be without this perk's contribution, and
-      // then take the difference between the total value and that to find the contribution.
-      const valueWithoutPerk = interpolateStatValue(itemStat.investmentValue - value, statDisplay);
-      value = itemStat.value - valueWithoutPerk;
-    } else if (itemStat) {
-      const valueWithoutPerk = Math.min(itemStat.investmentValue - value, itemStat.maximumValue);
-      value = itemStat.value - valueWithoutPerk;
+  // The plug that is currently inserted into the socket
+  const activePlug = socket.plugged;
+
+  // We need to calculate the base investment stat value for the item so we can correctly
+  // interpolate each plugs effect on the item.
+  const baseItemInvestmentStats: { [statHash: number]: number } = {};
+
+  if (activePlug) {
```

Comment:
I'd love some more comments that explain how the active plug is being pulled out and treated separately.

Reply:
Yup, I'll come back and revisit this once @nev-r is finished with the stat rework. I don't think it will be needed though from the sounds of that change as this will likely be gone.

Reply:
Why would this be gone? This is about stats from plugs which still exist.

Reply:
More so changed, I was talking to nev yesterday and they are going to rework a bunch of it. I think I was the commits in the other branch already.

Reply:
i didnt re-logic much of this particular function, but we can get stats.ts changes settled then revisit

Reply:
Yup also just as an fyi, calculating the items base investment stats and then just looping over all the plugs would be cleaner but there is a few more loops involved. I went with this method, even though its not as intuitive, as it should be slightly more performant. Happy to revisit this and do it the other way if you two think it won't have much of an impact though.

Reply:
This is fine I just want enough comments to remember why it's weird

Reply:
Easy done, I will sort it out once nevs stuff is in